### PR TITLE
VKontakte email

### PR DESCRIPTION
**Pull request by kudmni**
Some providers (VKontakte) give the email inside access token, so lets take it sends

### CODE REVIEW

**Comment by ramsey** [comment]
Since providers are being removed in v1.0.0, does this make more sense to add to the third-party VKontakte provider repo rather than here? https://github.com/j4k/oauth2-vkontakte

**Comment by kudmni** [comment]
I am already using https://github.com/j4k/oauth2-vkontakte.
However, without this fix access token loses email

**Comment by ramsey** [comment]
I'm really not a fan of adding additional features just to support a single provider. Ideally, we should be flexible enough that these kinds of differences between provider responses can be handled at the provider class level.

I'll do some thinking and looking at the VKontakte provider to see what we need to do in this library to provide that level of flexibility to the down-stream providers.

**Comment by kudmni** [comment]
Alternative is to override method getAccessToken as in my pull request into j4k/oauth2-vkontakte:
https://github.com/j4k/oauth2-vkontakte/pull/3

**Comment by ramsey** [comment]
Yeah. That's a lot of code just to get the email. I'll need to take a look at the VKontakte API to see how they're doing things.

**Comment by kudmni** [comment]
Here is documentation, but it is in russian: https://vk.com/dev/auth_sites
And especially about the email:
"В результате выполнения данного запроса Ваш сервер получит вновь созданный access_token. Вместе с access_token серверу возвращается время жизни ключа expires_in в секундах.
Также сервер передаёт email пользователя, если приложение является сайтом, были запрошены соответствующие права, а также у пользователя есть email, и он разрешил доступ к нему.
Процедуру авторизации приложения необходимо повторять в случае истечения срока действия access_token, смены пользователем своего логина или пароля или удаления приложения из настроек."

In a nutshell, the email is sent together with access token.

**Comment by ramsey** [comment]
Since user data is specific to the individual providers, I've decided to leave that up to each provider to handle. See discussion on #265 for reference—specifically here: https://github.com/thephpleague/oauth2-client/issues/265#issuecomment-94912180
